### PR TITLE
TUG in simulation

### PR DESCRIPTION
**Pull request by vvasco**
This PR introduces a `gazebo` plugin to handle a human actor in `gazebo`. 
The plugin allows the user to handle the animations associated with the actor, defined in a `.world` file. To simulate the TUG, the `tug-scenario.world` file includes the following animations:
- `stand`;
- `sit_down`;
- `sitting`;
- `stand_up`;
- `walk`.

The plugin exposes a `rpc` port, whose name is defined in `tugInterface.ini`. 
The user can:
 - use the rpc service `play` to play the sequence of animations (the order is defined in the `.world` file, specified by `trajectory id`). If we specify the name of the animation, it's possible to:
    1. play the single animation;
    2. play the script from the specified animation.

_Note: 1. and 2. are based on [this fork](https://bitbucket.org/vvasco/gazebo/commits/823708f55bc6c4ed471e1420995370cdaec9b230) of the gazebo libraries (I will open a PR)._

To simulate the TUG, we have the following scenario:
whole-tug-opt

- the user can set the target to reach during walking by:
    - specifying `targets` in `tugInterface.ini`, which is a matrix containing the poses to reach in the form `x y theta`;

    - specifying the location to reach, in the form `x y theta`, using the `goTo` service: 

go-to

- use the rpc service `setSpeed` to set the walking speed during a walking animation. 

|             |          |
|-----------|--------|
|  slower |  slower-opt |
|  faster |  faster  |

### CODE REVIEW

**Comment by traversaro** [comment]
>  Note: 1. and 2. are based on this fork of the gazebo libraries (I will open a PR).

**Comment by vvasco** [comment]
> > Note: 1. and 2. are based on this fork of the gazebo libraries (I will open a PR).
> 
> You are probably aware of this, but Gazebo recently migrated to GitHub (see https://community.gazebosim.org/t/important-gazebo-and-ignition-are-going-to-github/533/4) so the Gazebo repo is available at https://github.com/osrf/gazebo .

Thank you @traversaro, I didn't know.

File: app/gazebo/tug/plugins/tuginterface/interface/include/tugserver.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#include <yarp/sig/Matrix.h>
+
+#include <gazebo/gazebo.hh>
+#include <gazebo/physics/World.hh>
+#include <gazebo/physics/Actor.hh>
+
+#include <include/utils.h>
+#include <../include/TugInterfaceServer.h>
+#include <../include/Pose.h>
+#include <../include/Animation.h>
```
Comment (pattacini) [inline app/gazebo/tug/plugins/tuginterface/interface/include/tugserver.h:21]: Assuming the availability of relative paths is not that good.
Better off arranging for exposing proper paths instead through CMake.

Comment (vvasco) [reply app/gazebo/tug/plugins/tuginterface/interface/include/tugserver.h]: I updated the CMakeLists and removed relative paths.

**Comment by vvasco** [comment]
> Perhaps, it'd be worth addressing also @traversaro's comment at this stage and thus pointing to the latest Gazebo resources on GitHub.

I'm going to update the changes I made on the GitHub repo and open the related PR there.

File: app/gazebo/tug/plugins/tuginterface/interface/CMakeLists.txt [hunk 2]
```diff
@@ -6,14 +6,14 @@ set(gazebo_assistiverehab_tuginterface_source src/tuginterface.cpp src/tugserver
 set(gazebo_assistiverehab_tuginterface_headers include/tuginterface.h include/tugserver.h include/utils.h)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}
```
Comment (pattacini) [inline app/gazebo/tug/plugins/tuginterface/interface/CMakeLists.txt:8]: @vvasco you may consider using `target_include_directories` instead, which is newer.

File: app/gazebo/tug/plugins/tuginterface/interface/CMakeLists.txt [hunk 3]
```diff
@@ -6,14 +6,14 @@ set(gazebo_assistiverehab_tuginterface_source src/tuginterface.cpp src/tugserver
...
+                    ${TIRPC_HEADERS}
                     ${Boost_INCLUDE_DIRS}
                     ${GAZEBO_INCLUDE_DIRS})
 link_directories(${GAZEBO_LIBRARY_DIRS})
```
Comment (pattacini) [inline app/gazebo/tug/plugins/tuginterface/interface/CMakeLists.txt:12]: Is this really required? Asking 'cause `link_directories` is quite old-fashioned cmake stuff.

Comment (vvasco) [reply app/gazebo/tug/plugins/tuginterface/interface/CMakeLists.txt:12]: I found it in the tutorial [here](http://gazebosim.org/tutorials?tut=plugins_hello_world). 
I think that linking to `${GAZEBO_LIBRARIES}` through `target_link_libraries()` is sufficient and we can remove the `link_directories`.

Comment (pattacini) [reply app/gazebo/tug/plugins/tuginterface/interface/CMakeLists.txt:12]: Just double check it to make sure that this is sufficient.

Comment (vvasco) [reply app/gazebo/tug/plugins/tuginterface/interface/CMakeLists.txt:12]: Yes I double checked and it's sufficient.

**Pull request merged**